cmd/capabilities/dispatcher/dispatcher.go

```diff
@@ -0,0 +1,169 @@
+        "github.com/apigee/registry/cmd/capabilities/utils"
+        cloudtasks "cloud.google.com/go/cloudtasks/apiv2"
+        taskspb "google.golang.org/genproto/googleapis/cloud/tasks/v2"
+)
+
+// Make this configurable
+const topicName = "changes"
+const subscriptionName = topicName + "-pull-subscriber"
+
+func main() {
+    log.Print("Starting subscriber...")
+    ctx := context.Background()
+
+    // Setup and start the dispatcher server
+    dispatcher := &Dispatcher{}
+    err := dispatcher.setUp(ctx)
+    if err != nil {
+        log.Printf(err.Error())
+        return
+    }
+    err = dispatcher.startServer(ctx)
+    if err != nil {
+        log.Printf(err.Error())
+    }
+    return
+}
+
+type Dispatcher struct {
+    pubsubClient *pubsub.Client
+    subscription *pubsub.Subscription
+}
+
+func (d *Dispatcher) setUp(ctx context.Context) error {
+    var err error
+    // Create pubsub client
+    d.pubsubClient, err = pubsub.NewClient(ctx, os.Getenv("REGISTRY_PROJECT_IDENTIFIER"))
+    if err != nil {
+        return err
+    }
+
+    // Create topic
+    var topic *pubsub.Topic
+    topic, err = d.pubsubClient.CreateTopic(ctx, topicName)
+    if err != nil {
+        code := status.Code(err)
+        if code == codes.AlreadyExists {
+            topic = d.pubsubClient.Topic(topicName)
+        } else {
+            return err
+        }
+    }
+
+    // Create subscription
+    d.subscription, err = d.pubsubClient.CreateSubscription(ctx, subscriptionName, pubsub.SubscriptionConfig{
+            Topic:            topic,
+            AckDeadline:      10 * time.Second,
+        })
+    if err != nil {
+        code := status.Code(err)
+        if code == codes.AlreadyExists {
+            d.subscription = d.pubsubClient.Subscription(subscriptionName)
+        }
+    }
+    log.Printf("%s", d.subscription)
+
+    // Configure subscriber.
+    // TODO: Make this configurable.
+    d.subscription.ReceiveSettings.MaxOutstandingMessages = 10
+    d.subscription.ReceiveSettings.NumGoroutines = runtime.NumCPU()
+    return nil
+}
+
+func (d *Dispatcher) startServer(ctx context.Context) error {
+    err := d.subscription.Receive(ctx, messageHandler)
+    if err != nil {
+        return err
+    }
+    return nil
+}
+
+func messageHandler(ctx context.Context, msg *pubsub.Message) {
+    data := string(msg.Data)
+    message := rpc.Notification{}
+    if err := jsonpb.UnmarshalString(data, &message); err != nil {
+        log.Printf("Message data: %s", data)
+        log.Printf("Error in json.Unmarshal: %v", err)
+        msg.Ack()
+        return
+    }
+
+    switch changeType := message.Change; changeType {
+        case rpc.Notification_CREATED, rpc.Notification_UPDATED:
+            log.Printf("running task for change type %q", changeType)
+        default:
+            log.Printf("ignoring change type %q", changeType)
+            msg.Ack()
+            return
+    }
+
+    err := createQueueTask(ctx, message.Resource)
```

Comment:
Is this verifying that the changed resource is a spec? (if so, I missed that - if not, it seems that it would trigger linter runs for new APIs, versions, and even artifacts)

Reply:
I had planned to take care of this through config maps but that will come in a separate PR. Added checks the type of resource.